packages/main/src/components/AnalyticalTable/index.tsx

```diff
@@ -248,6 +251,48 @@ const AnalyticalTable: FC<TableProps> = forwardRef((props: TableProps, ref: Ref<
 
   const [headerRef, tableWidth] = useWindowResize();
 
+  const [cols, setCols] = useState(columns);
```

Comment:
could you extract the drag and drop logic to a custom `useDragAndDrop` hook like e.g. `useWindowResize`?